src/app/shared/components/template/components/button/button.component.html

```diff
@@ -1,18 +1,50 @@
-<ion-button
-  [class]="'full standard medium' + ' ' + style + ' ' + buttonAlign"
-  [disabled]="disabled"
-  (click)="triggerActions('click')"
-  [attr.data-param-style]="style"
->
-  <ion-icon *ngIf="icon" slot="start" src="{{ icon | plhAsset }}"></ion-icon>
-  <span *ngIf="_row.value" [class]="textAlign" [innerHTML]="_row.value | markdown"> </span>
-  <span *ngIf="_row.rows" class="children">
-    <plh-template-component
-      *ngFor="let childRow of _row.rows | filterDisplayComponent; trackBy: trackByRow"
-      class="child"
-      [row]="childRow"
-      [parent]="parent"
-      [attr.data-rowname]="_row.name"
-    ></plh-template-component>
-  </span>
-</ion-button>
+<ng-container [ngSwitch]="variantMap.cardPortrait">
```

Comment:
I couldn't think of a cleverer way of using `ngSwitch` with the `variantMap` than this, which would require additional nested `ngSwitch` statements if we add support for additional elementary variants. But let me know if I'm missing something

Reply:
Ah, yeah I see the issue. Really we want a slightly more sophisticated system where as part of the variant mapping system we identify specific base template (with multiple html templates defined either in one file or multiple).

I'd probably recommend tidying up at some point by adding another variable that tracks `templateBase` or similar, but not really urgent (maybe just next time working on the component if there's more variants to add that have structural changes)